doc/proposals/shared-memory-transport/eprosima/interprocess_shared_mem.md

```diff
@@ -0,0 +1,95 @@
+# Interprocess Shared Memory model
+
+## Overview
+
+This document describes a model for interprocess shared memory communication  of software components interchanging data messages. The objective of the model is to provide a shared memory based transport layer suitable for a Real-time Publish-Subscribe DDS (Data Distribution Service).
+
+## Context
+
+eProsima has received commercial proposals to implement a shared memory transport as an improvement of its FastRPTS product. The goals listed in this document are extracted from the customers proposals.
+
+### Improvements over the standard network transports
+
+* Reduce OS Kernel calls: This is unavoidable for transports like UDP / TCP (even with the loopback interface).
+* Large message support: Network stacks needs to fragment large packets.
+* Avoid data serialization / deserialization process: This is not possible in heterogeneous networks, but is possible in shared memory interprocess communication.
```

Comment:
Although it would be possible to avoid serialization / deserialization, we are not addressing this issue, at least on this implementation (serialization is done outside the transport layer) We shouldn't mention it as an improvement if we are not going to do it.

Reply:
I still think is good to mention, sure the upper layer needs to be aware of that and implement different cases for serialized and not serialized transports. I think is similar to "large message support", the upper layer can query max message size to the transport and decide wheather or not fragmentation is needed. 

We could add a property "is_serialized_needed" for the upper layer to decide if serialization can be avoided.